### PR TITLE
Fix for removing checkpoints in wrong order

### PR DESCRIPTION
**Pull request by peterscherpelz**

### CODE REVIEW

File: mewarpx/mewarpx/utils_store/init_restart_util.py [hunk 1]
```diff
@@ -3,36 +3,94 @@
...
-default_checkpoint_name = "checkpoint"
 
-def clean_old_checkpoints(prefix=default_checkpoint_name, directory="diags",
-                          num_to_keep=0):
+def get_sorted_checkpoints(checkpoint_directory, checkpoint_prefix):
+    """Function to return valid checkpoints for restarting or removing
+    checkpoints.
+
+    Also warns if any checkpoints with .old. exist, which shouldn't in normal
+    workflow.
```
Comment (roelof-groenewald) [inline mewarpx/mewarpx/utils_store/init_restart_util.py:18]: At present we don't clean up an existing `diags` folder when starting a run as we do in `metools`. I think that would probably be a good addition to this PR - whenever a fresh run (not a restart) is started, we should delete `diags` if present. What do you think?

Comment (peterscherpelz) [reply mewarpx/mewarpx/utils_store/init_restart_util.py:18]: Cleanest way to do it we think:
- Determine `self.restart` where it is in the code.
- Between that and initialization, `if not self.restart` then remove all `diags*` directories
- At initialization, all diagnostics would create folders with an afterinit python hook. diags folders created before init would be incorrect programming.
Let's do this in a separate PR, I'll make a monday.com card.

**Review by roelof-groenewald — APPROVED** [review]

File: mewarpx/mewarpx/mwxrun.py [hunk 2]
```diff
@@ -299,7 +316,8 @@ def get_domain_area(self):
             return ((pos_lims[1] - pos_lims[0]) * (pos_lims[3] - pos_lims[2]))
         else:
             # TODO: implement RZ when needed
-            raise NotImplementedError("get_domain_area not implemented for RZ geometry")
+            raise NotImplementedError(
+                "get_domain_area not implemented for RZ geometry")
```
Comment (KZhu-ME) [inline mewarpx/mewarpx/mwxrun.py]:
```suggestion
            raise NotImplementedError(
                "get_domain_area not implemented for RZ geometry"
            )
```

**Review by KZhu-ME — APPROVED** [review]
Looks good! Just one style thing I found.

**Pull request merged**